### PR TITLE
Remove all #ifdef macros

### PR DESCRIPTION
**Pull request by middlecoff**
## Proposed changes

To reduce the size of the code, remove all `#ifdef` macros from `sw_core.F90`. None of the `#ifdef`s were used anywhere. 

## Types of changes

- [ ] Bugfix (Change which fixes an issue)
- [ ] New feature (Change which adds functionality)
- [x] Enhancement to existing functionality (non-breaking change which improves existing code or documentation)
- [ ] Breaking change (fix or feature that changes results and requires updates to test baselines)
- [ ] Documentation Update

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/NOAA-GSL/SENA-c_sw/blob/develop/CONTRIBUTING.md) document
- [x] I have ensured all my changes contribute to a common purpose
- [x] I have verified my change does not add debug code
- [x] I have verified my change does not add code that is commented out
- [x] I have verified unit tests pass locally with my changes
- [x] I have verified my changes adhere to style guidelines
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by christopherwharrop-noaa — APPROVED** [review]
This is a clean removal of the `#ifdef` macros.  No impact on the code paths that are actually used, and no impact to code style.

File: ref/src/sw_core.F90
Comment (LyndStringer) [inline ref/src/sw_core.F90:129]: I thought this was just removing #ifdef(s).  This is a change to an interface.

File: ref/src/sw_core.F90
Comment (LyndStringer) [inline ref/src/sw_core.F90:368]: Interface change.

**Review by LyndStringer — CHANGES_REQUESTED** [review]
I don't think we should be changing interfaces.  Please check with Chris.  I'm OK with this if Chris is OK with this.

File: ref/src/sw_core.F90
Comment (middlecoff) [reply ref/src/sw_core.F90:129]: The variables removed form the interface were only used in the IFDEF'd part of the code so when the IFDEFs were removed those variables resulted in a compiler warning that they were never referenced so I thought it would be cleaner to remove the variables.

Comment (christopherwharrop-noaa) [reply ref/src/sw_core.F90:129]: @LyndStringer - I also noticed the change in interface, but concluded that it was correct since those variables are no longer used in that subroutine.  This boils down to the same question that is motivating these changes in the first place.  We are removing dead (for our use case) code.  If we've decided that we're only going to keep what is actually on the code path our use case follows, then I think this cleanup fits that goal.  If we start saying we'll keep "this" dead code but throw away "that" dead code, then it's hard to know what the guiding principle/goal is.  I say remove it.

**Review by robgpita — APPROVED** [review]
I'll defer to everyone else's expertise on this one. The more dead code that is removed, the easier my job is...

**Comment by christopherwharrop-noaa** [comment]
@LyndStringer - Do you wish to have more discussion about the change to the subroutine interface?

File: ref/src/sw_core.F90
Comment (middlecoff) [reply ref/src/sw_core.F90:129]: Since Lynd agrees, this issue is resolved by leaving the questioned changes as they were.

**Comment by christopherwharrop-noaa** [comment]
@LyndStringer - Since you requested changes, this PR cannot be merged until you approve it.  Please either approve or further explain your request for changes to continue the discussion.

File: ref/src/sw_core.F90
Comment (LyndStringer) [reply ref/src/sw_core.F90:129]: My bad, I thought Jacques could resolve this conversation.  I'll resolve it immediately.

**Review by LyndStringer — APPROVED** [review]
Looks good.  Passes the tests.  Jacques, now many lines were removed?

**Comment by middlecoff** [comment]
Counting all three pull requests, 324 lines were removed which is about 20%.

**Pull request merged**
